sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -306,11 +296,9 @@ class SimpleFunctionRegistry
     extends SimpleFunctionRegistryBase[Expression]
     with FunctionRegistry {
 
-  override def clone(): SimpleFunctionRegistry = synchronized {
+  override def clone(): SimpleFunctionRegistry = {
```

Comment:
Originally, this entire function was atomic. If there are additions or deletions to functionBuilders during the cloning process, it would deviate from the previous behavior.

Reply:
Yes. I know it.
ConcurrentHashMap creates the iterator with weakly consistent and it will not causes concurrent modification exception.
I think the clone for SimpleFunctionRegistry allows the concurrent modification.

Reply:
My question is, is it possible that modifications to `functionBuilders` during the cloning process could result in the cloned object not being an exact snapshot of the original at that time? For example, could it have additional functions or be missing certain functions?

Reply:
it could promises clone once each still-valid node that was reachable upon iterator construction. 
After the construction, the modification will not affect the cloned instance.